packages/pages-shared/asset-server/handler.ts

```diff
@@ -585,15 +604,66 @@ export async function generateHandler<
 
 	async function notFound(): Promise<Response> {
 		if (caches) {
-			const assetPreservationCache = await caches.open(
-				ASSET_PRESERVATION_CACHE
-			);
-			const preservedResponse = await assetPreservationCache.match(request.url);
-			if (preservedResponse) {
-				if (setMetrics) setMetrics({ preservationCacheResult: "checked-hit" });
-				return preservedResponse;
-			} else {
-				if (setMetrics) setMetrics({ preservationCacheResult: "checked-miss" });
+			try {
+				const assetPreservationCacheV2 = await caches.open(
+					ASSET_PRESERVATION_CACHE_V2
+				);
+				let preservedResponse = await assetPreservationCacheV2.match(
+					getCacheKey()
+				);
+
+				// Continue serving from V1 preservation cache for some time to
+				// prevent 404s during the migration to V2
+				const cutoffDate = new Date("2024-05-10");
+				if (!preservedResponse && Date.now() < cutoffDate.getTime()) {
+					const assetPreservationCacheV1 = await caches.open(
+						ASSET_PRESERVATION_CACHE_V1
+					);
+					preservedResponse = await assetPreservationCacheV1.match(request.url);
+					if (preservedResponse) {
+						// V1 cache contains full response bodies so we return it directly
+						if (setMetrics) {
+							setMetrics({ preservationCacheResult: "checked-hit" });
+						}
+						return preservedResponse;
```

Comment:
Fixed this logic - the V1 cache contains asset bodies so we need to serve the response directly